core/services/keystore/workflow_encryption.go

```diff
@@ -0,0 +1,162 @@
+package keystore
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/workflowencryptionkey"
+)
+
+// ErrWorkflowEncryptionKeyExists describes the error when the workflow encryption key already exists
+var ErrWorkflowEncryptionKeyExists = errors.New("can only have 1 Workflow encryption key")
+
+type WorkflowEncryption interface {
+	Get(id string) (workflowencryptionkey.Key, error)
+	GetAll() ([]workflowencryptionkey.Key, error)
+	Create(ctx context.Context) (workflowencryptionkey.Key, error)
+	Add(ctx context.Context, key workflowencryptionkey.Key) error
+	Delete(ctx context.Context, id string) (workflowencryptionkey.Key, error)
+	Import(ctx context.Context, keyJSON []byte, password string) (workflowencryptionkey.Key, error)
+	Export(id string, password string) ([]byte, error)
+	EnsureKey(ctx context.Context) error
+}
+
+type workflowEncryption struct {
+	*keyManager
+}
+
+var _ WorkflowEncryption = &workflowEncryption{}
+
+func newWorkflowEncryptionKeyStore(km *keyManager) *workflowEncryption {
+	return &workflowEncryption{
+		km,
+	}
+}
+
+func (ks *workflowEncryption) Get(id string) (workflowencryptionkey.Key, error) {
```

Comment:
is it possible to return pointer and return nil in case of error?

Reply:
I initially did it like this, but if you check the rest of the keystore implementations they all return a copy instead of a pointer. For simplicity and consistency I choose that, do you see a performance improv or is it just to be able to return nil on error?

Reply:
ok, let's keep the copy